vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridTestbenchUtilIT.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2000-2023 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.grid.it;
+
+import com.vaadin.flow.component.grid.testbench.GridColumnElement;
+import com.vaadin.flow.component.grid.testbench.GridElement;
+import com.vaadin.flow.component.grid.testbench.GridTHTDElement;
+import com.vaadin.flow.component.grid.testbench.GridTRElement;
+import com.vaadin.flow.testutil.TestPath;
+import com.vaadin.tests.AbstractComponentIT;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.stream.IntStream;
+
+/**
+ * Tests for ensuring utility methods to fetch rows, and cells from specific
+ * rows and columns work as expected.
+ */
+@TestPath("vaadin-grid-it-demo/all-rows-visible")
+public class GridTestbenchUtilIT extends AbstractComponentIT {
+
+    private GridElement grid;
+
+    @Before
+    public void init() {
+        open();
+        grid = $(GridElement.class).first();
+    }
+
+    @Test
+    public void cellsRowIndexUtilMethod_returnsExpectedCells() {
+        IntStream.rangeClosed(grid.getFirstVisibleRowIndex(),
+                grid.getLastVisibleRowIndex()).forEach(row -> {
```

Comment:
This file seems to be from an earlier revision before [these comments](https://github.com/vaadin/flow-components/pull/5508#discussion_r1352095602) were addressed.

Reply:
Will take a look at this